saltproc/depcode.py

```diff
@@ -159,20 +166,346 @@ def write_mat_file(self, dep_dict, dep_end_time):
+    Contains methods to read template and output files,
+    write new input files for OpenMC.
+
+    Attributes:
+    -----------
+    param : dict of str to type
+        Holds depletion steo parameter information
+    sim_info : dict of str to type
+        Holds simulation settings information
+    iter_inputfile : dict of str to str
+        Paths to OpenMC input files for OpenMC rerunning.
+    iter_matfile : str
+        Path to iterative, rewritable material file for OpenMC
+        rerunning. This file is modified during the simulation.
+
+
+
+    """
+
+    def __init__(self,
+                 exec_path="openmc_deplete.py",
+                 template_inputfiles_path={"geometry": "./geometry.xml",
+                                           "materials": "./materials.xml",
+                                           "settings": "./settings.xml",
+                                           "chain_file": "./chain_simple.xml"},
+                 geo_files=None,
+                 npop=50,
+                 active_cycles=20,
+                 inactive_cycles=20):
+        """Initializes the DepcodeOpenMC object.
+
+           Parameters
+           ----------
+           exec_path : str
+               Path to OpenMC depletion script.
+           template_inputfiles_path : dict of str to str
+               Path to user input files (``.xml`` file for geometry,
+               material, and settings) for OpenMC.
+           iter_matfile : str
+               Name of iterative, rewritable material file for OpenMC
+               rerunning. This file is modified during  the simulation.
+           geo_files : str or list, optional
+               Path to file that contains the reactor geometry.
+               List of `str` if reactivity control by
+               switching geometry is `On` or just `str` otherwise.
+           npop : int, optional
+               Size of neutron population per cycle for Monte Carlo.
+           active_cycles : int, optional
+               Number of active cycles.
+           inactive_cycles : int, optional
+               Number of inactive cycles.
+
+        """
+        super().__init__("openmc",
+                         exec_path,
+                         template_inputfiles_path,
+                         geo_files,
+                         npop,
+                         active_cycles,
+                         inactive_cycles)
+        self.iter_inputfile = {'geometry': './geometry.xml',
+                               'settings': './settings.xml'},
+        self.iter_matfile = './materials.xml'
+
+    def read_depcode_info(self):
+        """Parses initial OpenMC simulation info from the OpenMC output files
+        and stores it in the `Depcode` object's ``sim_info`` attribute.
+        """
+
+    def read_depcode_step_param(self):
+        """Parses data from OpenMC depletion output for each step and stores
+        it in `Depcode` object's ``param`` attributes.
+        """
```

Comment:
Are these functions supposed to do anything? I don't see any text for them to do anything.

Reply:
They are uninplemented right now. They must be at least defined because they are abstract funtions in the `Depcode` abstract class.